### PR TITLE
make dist fixes, allow building in a minimal environment, re-enable packit

### PR DESCRIPTION
**Pull request by martinpitt**
This re-enables packit tests after commit c4908fe050dda.

This is a hack, but at least we have all our tests to tell us when we break this again.

- [ ] ~add some missing build dependencies to sandcastle container: https://github.com/packit/sandcastle/pull/122~

### CODE REVIEW

**Review by allisonkarlitskaya — COMMENTED** [review]
As we discussed on IRC: I'd prefer to try to construct an independent mock environment that makes `./configure` happy without making changes to our build system.  I'm confident that this is possible.

**Comment by martinpitt** [comment]
I moved that over to a bunch of mock .pc files and a single empty include. This isn't too bad IMHO, and at least nicely separated from our actual build system. The other commits are valid fixes IMHO.

File: tools/mock-build-env/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+These files are ...
```
Comment (martinpitt) [inline tools/mock-build-env/README.md:1]: ... still to be written, d'oh!

Comment (martinpitt) [reply tools/mock-build-env/README.md:1]: Fixed.

**Review by allisonkarlitskaya — APPROVED** [review]
I'm of two minds about your approach here:
 - on one hand, I feel like it would be better to mock up ssh/pcp/etc stuff and remove all the `--disable` in order to avoid running into a situation where we're testing a non-canonical build (since you noticed that we produced different results depending on our configure flags and such differences might slip in again in the future)
 - on the other hand, testing different scenarios is always good, and indeed, led you to find new bugs here.

So maybe a TODO item instead: you can merge this as is (or take my suggestion to add more "mocking" and less "disabling" and we do another round of review) and then later on we can add an explicit test scenario that tries to build cockpit with as much `--disable` as we can reasonably throw at it and checks to make sure we still get a viable result  which properly distchecks and so on...

**Comment by martinpitt** [comment]
For posterity, we discussed that on IRC:

    lis | but it's actually a good point that packit is basically doing exactly the testing i proposed

**Pull request merged**